catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/SubscriptionContainerImpl.java

```diff
@@ -0,0 +1,396 @@
+        CachedSubscription cachedSubscription = createCachedSubscription(metadata);
+        cachedSubscription.registerSubscription(subscription);
+        try {
+            subscriptionsCache.put(metadata.getId(), cachedSubscription);
+        } catch (CacheException e) {
+            cachedSubscription.unregisterSubscription();
+            throw new SubscriptionStoreException("Problem writing to the cache. ", e);
+        }
+    }
+
+    /**
+     * Helper method for core delete logic.
+     */
+    private CachedSubscription doDelete(String subscriptionId) {
+        CachedSubscription cachedSubscription = subscriptionsCache.get(subscriptionId);
+        try {
+            subscriptionsCache.remove(subscriptionId);
+        } catch (CacheException e) {
+            throw new SubscriptionStoreException("Problem deleting from cache. ", e);
+        }
+        cachedSubscription.unregisterSubscription();
+        return cachedSubscription;
+    }
+
+    /**
+     * Ensure the identifier points to a valid subscription, and that it exists in the container.
+     *
+     * @param operation customize the message based on where this validation is occurring.
+     */
+    private void validateContainment(SubscriptionIdentifier identifier, String operation) {
+        if (get(identifier) == null) {
+            LOGGER.debug("Target for subscription {} [ {} | {} ] does not exist",
+                    operation,
+                    identifier.getId(),
+                    identifier.getTypeName());
+            throw new SubscriptionStoreException(format("Subscription [%s] does not exist. ",
+                    identifier.getId()));
+        }
+    }
+
+    private void validateSubscription(Subscription subscription) {
+        notNull(subscription, "Subscription object cannot be null. ");
+    }
+
+    private void validateIdentifier(SubscriptionIdentifier identifier) {
+        notNull(identifier, "Subscription identifier cannot be null. ");
+        notEmpty(identifier.getId(), "Subscription ID string cannot be null or empty. ");
+        notEmpty(identifier.getTypeName(), "Subscription type string cannot be null or empty. ");
+
+        // TODO: Get Approval - Backslash is not the ECQL escape character, thus we need to forbid single quotes
+        if (identifier.getId()
+                .contains(SINGLE_QUOTE)) {
+            throw new SubscriptionStoreException("Subscription ID contains invalid characters");
+        }
+    }
+
+    private void validateType(SubscriptionType type) {
+        notNull(type, "Subscription type cannot be null. ");
+        notEmpty(type.getTypeName(), "Subscription type string cannot be null or empty. ");
+    }
+
+    private void validateSerialized(MarshalledSubscription marshalledSubscription) {
+        notNull(marshalledSubscription, "Serialized subscription cannot be null. ");
+        notEmpty(marshalledSubscription.getFilter(),
```

Comment:
❓ Can `MarshalledSubcription` enforce those constraints, i.e., no `null` or empty filter and callback address? If so, we should move those validations to that class.